### PR TITLE
Terraform aws rhel6 enhancement

### PR DESCRIPTION
**Pull request by sathiyas**
Enhanced the aws templates for consul set up to add rhel6. Made ubuntu as default so that no changes are needed for execution if ubuntu is needed. In future will submit PRs for rhel7 and centos versions. rhel7 has new way of doing upstarts, hence need to separate rhel6 and rhel7 implementations.

### CODE REVIEW

File: terraform/aws/variables.tf [hunk 1]
```diff
@@ -24,10 +38,18 @@ variable "servers" {
 }
 
 variable "instance_type" {
-    default = "m1.small"
-    description = "The instance type to launch."
+    default = {
+        ubuntu = "m1.small"
+        rhel6 = "m3.medium"
```
Comment (phinze) [inline terraform/aws/variables.tf]: Why a different default instance type for RHEL? Seems like we should just make them the same.

**Comment by phinze** [comment]
One question, otherwise this is looking good!

**Comment by sathiyas** [comment]
@phinze, thanks for asking, I should have mentioned it, RHEL AMIs are HVM based and do not take m1 instances.

**Comment by phinze** [comment]
@sathiyas ah okay - can you go t2.micro instead then? we want to keep the defaults in the examples relatively cheap :grinning: 

Ideally we can switch the ubuntu AMIs to HVM as well just to keep it simpler.

**Comment by phinze** [comment]
(Sorry accidentally hit the wrong button there)

**Comment by sathiyas** [comment]
Sure, let me update with both Ubuntu/RHEL6 HVM AMIs with t2.micro

**Comment by sathiyas** [comment]
@phinze , thanks for quick review, I made changes to reflect latest AMIs for HVM for both ubuntu and rhel6 and tested both.

**Comment by phinze** [comment]
Re-kicked the deploy - some ephemeral failures there. LGTM!

**Pull request merged**